src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,186 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        final Node<T> newTail = tail;
+        Node<T> newNode = new Node<>(tail, value, null);
+        tail = newNode;
```

Comment:
this variable is redundant, move `tail = newNode;` command after is/else construction and use tail itself instead of it's copy to new variable